aten/src/ATen/native/cuda/ReduceMomentKernel.cu

```diff
@@ -36,8 +36,8 @@ static void std_var_kernel_cuda(TensorIterator& iter, bool unbiased, bool take_s
 
 template <typename scalar_t, typename acc_t=scalar_t, typename out_t=scalar_t>
 void mean_kernel_impl(TensorIterator& iter) {
-  float factor = float(iter.num_output_elements()) / iter.numel();
-  gpu_reduce_kernel<scalar_t, out_t>(iter, MeanOps<acc_t, float> {factor});
+  scalar_t factor = scalar_t(iter.num_output_elements()) / iter.numel();
```

Comment:
what is this change for? I think factor should be float for all cases regardless of the scalar type

Reply:
@anjali411 This change is to handle the complex tensors.
If we make the `factor` float for all dtypes, there would be compiling errors in the `project` function as no overloaded versions of  `*` operator are found between `complex<double>` and `float` operands.
https://github.com/pytorch/pytorch/blob/eec201c1388434a8d1efbfaf3d6ab091247b9aa9/aten/src/ATen/native/SharedReduceOps.h#L142-L144

Compiling logs
```
E:/repo/ml/github/pytorch/aten/src\ATen/native/SharedReduceOps.h(143): error: no operator "*" matches these operands
            operand types are: c10::complex<double> * const float
          detected during:
            instantiation of "acc_t at::native::MeanOps<acc_t, factor_t>::project(acc_t) const [with acc_t=c10::complex<double>, factor_t=float]"
```